src/smart_contract.rs

```diff
@@ -1,15 +1,16 @@
 use crate::{
-    backend::{Backend, TxORecord},
-    error::Result,
+    backend::Backend,
+    error::{Result as NauResult, Error as NauError},
```

Comment:
Nit: I prefer leaving the core error as `Result` in the source code. Not a big deal either way though. I used `NauError` in the example code because from its perspective it's a foreign error.